search/src/main/java/com/connexta/search/query/QueryManagerImpl.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.search.query;
+
+import com.connexta.search.common.Index;
+import com.connexta.search.common.IndexCrudRepository;
+import com.connexta.search.query.exceptions.QueryException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.dao.DataAccessResourceFailureException;
+
+@Slf4j
+public class QueryManagerImpl implements QueryManager {
```

Comment:
We could make Spring to do the work of the QueryManager.
See https://spring.io/guides/gs/rest-hateoas/
Start reading at the part "So you simply create a plain old java object extending ResourceSupport..."

Reply:
Neat! I will look into that before the mob review.

Reply:
I think we should have Spring HATEOAS for another day. What you've done is fine.